packages/core/src/main/util/encryptedStore.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {DBSchema, IDBPDatabase, openDB} from 'idb';
+
+export interface IEncryptedStore {
+  saveSecretValue(primaryKey: string, value: Uint8Array): Promise<void>;
+
+  getsecretValue(primaryKey: string): Promise<Uint8Array | undefined>;
+}
+
+interface EncryptedDB extends DBSchema {
+  key: {
+    key: string;
+    value: CryptoKey;
+  };
+  secrets: {
+    key: string;
+    value: {
+      iv: Uint8Array;
+      encrypted: Uint8Array;
+    };
+  };
+}
+
+class EncryptedStore implements IEncryptedStore {
+  readonly #key: CryptoKey;
+  constructor(key: CryptoKey, private readonly db: IDBPDatabase<EncryptedDB>) {
+    this.#key = key;
+  }
+
+  async saveSecretValue(primaryKey: string, value: Uint8Array) {
+    const iv = await window.crypto.getRandomValues(new Uint8Array(12));
```

Comment:
should we think about our friends with a lack of trust for getRandomValues? 

also: should we consider a node case like we do in RandomUtil.ts?

Reply:
Talked with Sebastian, this is fine, they just need custom entropy value for identity generation :ok_hand: